### PR TITLE
Flow ANCM version number in the build output

### LINKED ISSUE #4016: Emit ANCM version information as part of the build
In 2.2, we can modify the version of the produced internal ANCM package- https://github.com/dotnet/versions/blob/master/build-info/dotnet/product/cli/release/2.2/build.xml#L20
We'll need a new approach for 3.0 which @natemcmaster will help define.

### LINKED ISSUE #7548: Upgrade IIS C++ projects to build with platformtoolset v142 (VS2019)
Depends on https://github.com/aspnet/AspNetCore/pull/7005

Let's update the C++ projects for ANCM to build with the Visual Studio 2019 toolset. This should be a matter of updating the .vcxproj and reacting to any warnings or errors the new toolset finds.
```
<PlatformToolset>v142</PlatformToolset>
```

### PR DESCRIPTION
**Pull request by analogrelay**
Fixes #4016 
Also fixes #7548 because I was getting build errors that told me I should update them to v142 anyway :)

This drops a file: `artifacts/installers/aspnetcoremodule.version` with contents like:

```
13.0.19101.0
[commit hash]
```

Still outstanding (need some assistance from @aspnet/build): working out how we could flow this forward to the final build output. This needs a little more understanding of Maestro/BAR than I have off-hand. I can add it to the manifest in [Maestro.csproj](https://github.com/aspnet/AspNetCore/blob/4964b338abe02bdbb21ff69928d38b11e85c90c9/build/Maestro/Maestro.csproj#L17) but I'm not sure of how that all flows to the final output drop.

### CODE REVIEW

File: eng/scripts/vs.buildtools.json [hunk 1]
```diff
@@ -17,8 +17,6 @@
         "Microsoft.VisualStudio.Component.NuGet.BuildTools",
         "Microsoft.VisualStudio.Component.VC.ATL",
         "Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
-        "Microsoft.VisualStudio.Component.VC.v141.ATL",
```
Comment (jkotalik) [inline eng/scripts/vs.buildtools.json]: IIRC there wasn't a v142 equivalent for this package. As long as things build correct, I think we are fine to remove this.

File: src/Servers/IIS/build/Build.Settings [hunk 2]
```diff
@@ -4,7 +4,7 @@
      <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">$(MSBuildThisFileDirectory)..\</SolutionDir>
      <Configuration Condition="'$(Configuration)'==''">Debug</Configuration>
      <Platform Condition="'$(Platform)' == ''">Win32</Platform>
-     <PlatformToolset>v141</PlatformToolset>
+     <PlatformToolset>v142</PlatformToolset>
```
Comment (jkotalik) [inline src/Servers/IIS/build/Build.Settings]: Ah, I think I missed this one before. When you open the IIS sln now, do you get a warning about needing to upgrade to v142 still?

File: src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/AspNetCore.vcxproj [hunk 3]
```diff
@@ -305,4 +305,12 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
+  <Target Name="AfterBuild">
+    <!-- Drop a file in the artifacts directory containing the ANCM version number -->
+    <ItemGroup>
+      <VersionFileContents Include="$(AspNetCoreModuleVersionMajor).$(AspNetCoreMinorVersion).$(AssemblyBuild).$(AspNetCorePatchVersion)" />
```
Comment (jkotalik) [inline src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/AspNetCore.vcxproj]: I'm going to want to verify these numbers tomorrow. I don't believe these are correct, but want to confirm before I correct you.

File: eng/scripts/vs.buildtools.json (hunk 1 above)
Comment (natemcmaster) [reply eng/scripts/vs.buildtools.json]: "Microsoft.VisualStudio.Component.VC.ATL" is the v142 equivalent, and it's listed already. :)

File: src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/AspNetCore.vcxproj [hunk 4]
```diff
@@ -305,4 +305,12 @@
...
+      <VersionFileContents Include="$(CommitHash)" />
```
Comment (natemcmaster) [inline src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/AspNetCore.vcxproj]:
```suggestion
      <VersionFileContents Include="$(SourceRevisionId)" />
```
CommitHash is going to be deprecated. SourceRevisionId is a well-known, built-in property in Microsoft.Common.targets available after the  InitializeSourceControlInformation target . Example usage:

https://github.com/aspnet/AspNetCore/blob/78bc2a10b9ea617269527f4091903471a299e197/eng/targets/CSharp.Common.targets#L9-L17

**Review by natemcmaster — COMMENTED** [review]
BAR doesn't support anything except NuGet packages right now, so there isn't really anything we can do with this file except upload it to our blob storage.

One minor tweak to make but otherwise looks fine

**Comment by anurse** [comment]
> except upload it to our blob storage.

Looking at the release pipeline, it appears that will happen automatically.

**Comment by anurse** [comment]
Looks like there is something broken about the C++ resolution with this fix:

```
TRACKER : error TRK0005: Failed to locate: "CL.exe". The system cannot find the file specified.
```

File: src/Servers/IIS/build/Build.Settings (hunk 2 above)
Comment (anurse) [reply src/Servers/IIS/build/Build.Settings]: > When you open the IIS sln now, do you get a warning about needing to upgrade to v142 still

You do not.

File: eng/scripts/vs.buildtools.json (hunk 1 above)
Comment (anurse) [reply eng/scripts/vs.buildtools.json]: Yep, I came to the same conclusion looking at the component directory: https://docs.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-community?view=vs-2019

File: src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/AspNetCore.vcxproj (hunk 4 above)
Comment (anurse) [reply src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/AspNetCore.vcxproj]: Awesome, I'm glad this is in `Microsoft.Common.targets` now!

**Review by natemcmaster — APPROVED** [review]

**Pull request merged**